--- svr-chansession.c	2020-05-27 01:29:43.809878689 +0300
+++ svr-chansession.c.new	2020-05-27 01:29:50.296673752 +0300
@@ -939,6 +939,8 @@
 	/* if we're debugging using valgrind etc, we need to keep the LD_PRELOAD
 	 * etc. This is hazardous, so should only be used for debugging. */
 #ifndef DEBUG_VALGRIND
+	char* container_env = NULL;
+	container_env = getenv("CONTAINER");
 #ifdef HAVE_CLEARENV
 	clearenv();
 #else /* don't HAVE_CLEARENV */
@@ -947,6 +949,9 @@
 		environ[0] = NULL;
 	}
 #endif /* HAVE_CLEARENV */
+	if (container_env) {
+		setenv("CONTAINER", container_env, 1);
+	}
 #endif /* DEBUG_VALGRIND */
 
 #if DROPBEAR_SVR_MULTIUSER
